File: Frontend/DiveApp/Services/UserService.swift
import Foundation

final class UserService {
    private let client: APIClient

    init(client: APIClient) {
        self.client = client
    }

    func fetchCurrentUser() async throws -> CurrentUser {
        try await client.send(Endpoint(path: "/api/v1/users/me"))
    }

    func setupProfile(nickname: String, profileImageURL: String?, phoneNumber: String?) async throws -> UserProfileResponse {
        let endpoint = Endpoint(
            path: "/api/v1/users/me/profile",
            method: .post,
            body: ProfileSetupRequest(nickname: nickname, profileImageURL: profileImageURL, phoneNumber: phoneNumber)
        )
        return try await client.send(endpoint)
    }
}

File: Web/src/services/userService.ts
import { apiClient } from '../core/network/apiClient';
import type { ProfileSetupRequest, CurrentUser, UserProfileResponse } from '../models/userModels';

export const userService = {
  async fetchCurrentUser(): Promise<CurrentUser> {
    const { data } = await apiClient.get<CurrentUser>('/api/v1/users/me');
    return data;
  },

  async setupProfile(body: ProfileSetupRequest): Promise<UserProfileResponse> {
    const { data } = await apiClient.post<UserProfileResponse>('/api/v1/users/me/profile', body);
    return data;
  },
};
